tests/unit/test_parser.py

```diff
@@ -80,3 +80,31 @@ def test_duplicate_line():
     )
     with pytest.raises(IndexError, match=r".*This error is likely from a duplicate line detected.*"):
         compliance.parser_map["cisco_ios"](logging).config_lines  # pylint: disable=expression-not-assigned
+
+
+def test_banner_delimiter_iosxr():
```

Comment:
These tests may be good, but hard for me to understand, can you add to a mock test here: https://github.com/networktocode/netutils/tree/develop/tests/unit/mock/config/parser/base/cisco_iosxr